frontend/src/pushapi.js

```diff
@@ -21,49 +17,16 @@ angular.module('clouway-push', [])
     };
 
     /**
-     * Set a method to call for opening of connection.
-     * The method must return a promise.
+     * Sets the backend url connection.
      *
-     * @param {function(subscriber)} method method that will be called for opening of connection.
+     * @param url the new url
      * @returns {*} the provider instance for chaining purposes.
      */
-    this.openConnectionMethod = function (method) {
-      this.connectionMethods.connect = method;
+    this.backendServiceUrl = function(url) {
```

Comment:
Another suggestion:
In what was `pushapi-mock.js` add inside the config block:
```js
.config(function ($provide, $httpBackend, pushApiProvider) {
  $httpBackend.whenGET(pushApiProvider.endpoints.serviceUrl).respond(200, "dummy-token");
  $httpBackend.whenPUT(pushApiProvider.endpoints.serviceUrl).respond(200, "");
  ...
  
  $provide.decorator('pushApi', function ($delegate) {
      $delegate.flush = $httpBackend.flush;
      return $delegate;
    });
})
```
and so in tests of user of pushApi:
```js
it('test case', function () {
  /* calling logic that uses pushApi methods ... */
  pushApi.flush();

  expect(...).toBe(...);
});
```